lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -233,15 +233,20 @@ module InternalConstants
     Fluent::Plugin.register_output('google_cloud', self)
 
     PLUGIN_NAME = 'Fluentd Google Cloud Logging plugin'.freeze
-    # Extract plugin version by finding the spec this file was loaded from.
+
     PLUGIN_VERSION = begin
+      # Extract plugin version from file path.
+      match_data = __FILE__.match(
+        %r{fluent-plugin-google-cloud-(?<version>[0-9a-zA-Z\.]*)/})
+      return match_data['version'] if match_data
```

Comment:
I don't think you can `return` from a `begin`/`end` block... This is the pitfall of having different behaviors in unit tests and at runtime.

Reply:
Updated